app/src/main/java/com/hedvig/app/feature/settings/SettingsActivity.kt

```diff
@@ -37,9 +38,21 @@ import org.koin.android.viewmodel.ext.android.viewModel
 
 class SettingsActivity : BaseActivity(R.layout.activity_settings) {
     private val binding by viewBinding(ActivitySettingsBinding::bind)
+    private val marketProvider: MarketProvider by inject()
 
+    @SuppressLint("ApplySharedPref")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        val previousLanguage = PreferenceManager.getDefaultSharedPreferences(this).getString(SETTING_LANGUAGE, null)
```

Comment:
Couldn't we do this in `HedvigApplication::onCreate`? 🤔 That way, everyone will get migrated from the legacy setting to a supported setting, right? 🤔

Reply:
That is much better, ill fix it